src/vs/workbench/contrib/positronConnections/browser/components/schemaNavigation.tsx

```diff
@@ -0,0 +1,301 @@
+	}, [context.reactComponentContainer, scrollStateRef, setScrollState]);
+
+	const [selectedId, setSelectedId] = useState<string>();
+	const activeInstance = context.connectionsService.getConnections().find(item => item.id === activeInstanceId);
+
+	const [entries, setEntries] = useState<IPositronConnectionEntry[]>(activeInstance?.getEntries() || []);
+
+	useEffect(() => {
+		if (!activeInstance) {
+			return;
+		}
+		const disposableStore = new DisposableStore();
+
+		disposableStore.add(activeInstance.onDidChangeEntries((entries) => {
+			setEntries(entries);
+		}));
+
+		disposableStore.add(activeInstance.onDidChangeStatus((active) => {
+			if (!active) {
+				setActiveInstanceId(undefined);
+			}
+		}));
+
+		activeInstance.refreshEntries();
+		return () => disposableStore.dispose();
+	}, [activeInstance, setActiveInstanceId]);
+
+	if (!activeInstance) {
+		// This should not be possible, the active instance must exist.
+		return (
+			<div className='positron-connections-schema-navigation'>
+				<ActionBar
+					{...context}
+					disconnectHandler={() => { }}
+					refreshHandler={() => { }}
+					backHandler={() => props.setActiveInstanceId(undefined)}
+				>
+				</ActionBar>
+			</div>
+		);
+	}
+
+	const toggleExpandHandler = (id: string) => {
+		activeInstance.onToggleExpandEmitter.fire(id);
+	};
+
+	const ItemEntry = (props: ItemEntryProps) => {
+		const itemProps = entries[props.index];
+
+		return (
+			<PositronConnectionsItem
+				item={itemProps}
+				selected={itemProps.id === selectedId}
+				onSelectedHandler={() => setSelectedId(itemProps.id)}
+				onToggleExpandHandler={toggleExpandHandler}
+				style={props.style}>
+			</PositronConnectionsItem>
+		);
+	};
+
+	const { name, language_id, icon } = activeInstance.metadata;
+	const DETAILS_BAR_HEIGHT = 26;
+
+	return (
+		<div className='positron-connections-schema-navigation'>
+			<ActionBar
+				{...context}
+				disconnectHandler={() => activeInstance.disconnect?.()}
+				refreshHandler={() => activeInstance.refresh?.()}
+				backHandler={() => props.setActiveInstanceId(undefined)}
+			>
+			</ActionBar>
+			<div className='connections-items-container'>
+				<div className={'connections-instance-details'} style={{ height: DETAILS_BAR_HEIGHT }}>
+					<div className='connection-name'>{name}</div>
+					<div className='connection-language'>{languageIdToName(language_id)}</div>
+					<div className={'connection-icon'}>
+						{
+							icon || <div className='codicon codicon-positron-database-connection'></div>
+						}
+					</div>
+				</div>
+				<List
+					itemCount={entries.length}
+					itemSize={26}
+					/* size if the actionbar and the secondary side bar combined) */
+					height={height - ACTION_BAR_HEIGHT - DETAILS_BAR_HEIGHT}
+					width={'calc(100% - 2px)'}
+					itemKey={index => entries[index].id}
+					innerRef={innerRef}
+				>
+					{ItemEntry}
+				</List>
+			</div>
+		</div>
+	);
+};
+
+interface ItemEntryProps {
+	index: number;
+	style: any;
```

Comment:
Better as `style: CSSProperties | undefined;` (or without the `undefined`, if style is required).

Reply:
Resolved in 75d5390dbdbcaf6ec923f296646ad6dab6f034b2